server/project/project.go

```diff
@@ -412,7 +412,8 @@ func (s *Server) Update(ctx context.Context, q *project.ProjectUpdateRequest) (*
 		destCluster, err := argo.GetDestinationCluster(ctx, a.Spec.Destination, s.db)
 		if err != nil {
 			if err.Error() != argo.ErrDestinationMissing {
-				return nil, err
+				// If cluster is not found, we should discard this app, as it's most likely already in error
```

Comment:
For an application that is deploying to a removed cluster by server, wouldn't [this](https://github.com/argoproj/argo-cd/blob/e692a22b0114474360a31fbcbe297d7317f2dd5e/util/argo/argo.go#L1035) be the returned error?